pyface/util/color_parser.py

```diff
@@ -0,0 +1,441 @@
+
+
+def parse_hex(text):
+    """ Parse a hex form of a color.
+
+    Parameters
+    ----------
+    text : str
+        A string holding a hex representation of the color in the form
+        '#RGB', '#RGBA', '#RRGGBB', '#RRGGBBAA', '#RRRRGGGGBBBB', or
+        '#RRRRGGGGBBBBAAAA'.
+
+    Returns
+    -------
+    result : (space, channels), or None
+        Either a tuple of the form (space, channels), where space is one of
+        'rgb', 'rgba', 'hsv', 'hsva', 'hls', or 'hlsa' and channels is a
+        tuple of 3 or 4 floating point values between 0.0 and 1.0, includive;
+        or None if no functional representation could be matched.
+    """
+    text = text.strip()
+    if re.match("#[0-9a-fA-F]+", text) is None:
+        return None
+    text = text[1:]
+    if len(text) in {3, 4}:
+        step = 1
+    elif len(text) in {6, 8}:
+        step = 2
+    elif len(text) in {12, 16}:
+        step = 4
+    else:
+        return None
+    maximum = (1 << 4 * step) - 1
+    channels = ints_to_channels(
+        (int(text[i:i+step], 16) for i in range(0, len(text), step)),
+        maximum=maximum,
+    )
+    space = 'rgb' if len(channels) == 3 else 'rgba'
+    return space, channels
+
+
+class ColorParseError(ValueError):
+    """ An Exception raised when parsing fails. """
+    pass
+
+
+def parse_text(text):
+    """ Parse a text representation of a color.
+
+    Parameters
+    ----------
+    text : str
+        A string holding the representation of the color.  This can be:
+
+        - a color name, including all CSS color names, plus any additional
+          names found in pyface.color.color_table.  The names are normalized
+          to lower case and stripped of whitespace, hyphens and underscores.
+
+        - a hex representation of the color in the form '#RGB', '#RGBA',
+          '#RRGGBB', '#RRGGBBAA', '#RRRRGGGGBBBB', or '#RRRRGGGGBBBBAAAA'.
+
+        - a CSS functional representation, including "rgb()", "rgba()",
+          "hsv()", "hsva()", "hls()", "hlsa()".  Channel values are expected
+          to be in the range 0.0 to 1.0, inclusive.
+
+        - a tuple of rgb or rgba values, eg. '(0.6, 0.2, 0.4, 1.0)'.  Channel
+          values are expected to be in the range 0.0 to 1.0, inclusive.
+
+    Returns
+    -------
+    space : str
+        A string describing the color space for the channels.  Will be one of
+        'rgb', 'rgba', 'hsv', 'hsva', 'hls', 'hlsa'.
+    channels : tuple of floats
+        The channel values as a tuple of 3 or 4 floating point values between
+        0.0 and 1.0, includive.
+
+    Raises
+    ------
+    ColorParseError
+        If the string cannot be converted to a valid color.
+    """
+    result = None
+    for parser in parse_name, parse_functional, parse_hex:
+        result = parser(text)
+        if result is not None:
+            return result
+    else:
+        raise ColorParseError(
+            'Unable to parse color value in string {!r}'.format(text)
+        )
```

Comment:
If we replace string parsing with using Python functions directly, we will be able to get better error response.
cf. point 4 and the final proposal of my review comment.